### PR TITLE
Move all encoding function to util library

### PR DESCRIPTION
**Pull request by bonnetn**
This PR move all the encoding functions (encode/decode) to the util package.

Also, I defined an encode and a decode function for each one of the packet type.

### CODE REVIEW

File: src/util/encoding/packet.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+"""
+import struct
+from dataclasses import dataclass
+
+HEARTBEAT = struct.Struct('<B')
+PROPULSION_MOVEMENT_DONE = struct.Struct('<B')
+PROPULSION_ENCODER_POSITION = struct.Struct('<ii')
+PROPULSION_MOVE_WHEELS = struct.Struct('<ii')
+PROPULSION_STOP = struct.Struct('<')
+SERVO_SET_ANGLE = struct.Struct('<BBB')
```
Comment (tictacmenthe) [inline src/util/encoding/packet.py]: Maybe use a prefix for these "constants" so they are easy to find later? (PACKET_... or something)

Comment (tictacmenthe) [reply src/util/encoding/packet.py]: Also, if this is done, add a prefix or class to the classes below, and stay consistant with it.
Ex: PropulsionEncoderPositionPacket is fine, but PropulsionMovementDone doesn't follow the same rule.

File: src/util/encoding/packet.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    """
+    return PROPULSION_STOP.pack()
+
+
+def decode_propulsion_stop(_: bytes) -> PropulsionStopPacket:
+    """
+    Decode a stop message.
+    """
+    return PropulsionStopPacket()
+
```
Comment (tictacmenthe) [inline src/util/encoding/packet.py]: All this feels a bit pointless, isn't there a better way to do this all?

File: src/util/encoding/packet.py (hunk 1 above)
Comment (bonnetn) [reply src/util/encoding/packet.py]: Done thanks!

File: src/util/encoding/packet.py (hunk 2 above)
Comment (bonnetn) [reply src/util/encoding/packet.py]: Just for consistency, let's keep it here for now and remove in the future if we have nothing to add there.

**Pull request merged**